presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -550,8 +550,9 @@ public PlanOptimizers(
                         statsCalculator,
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(
-                                new ApplyTableScanRedirection(metadata),
-                                new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
+                                new PushProjectionIntoTableScan(metadata, typeAnalyzer),
+                                new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer),
+                                new ApplyTableScanRedirection(metadata))));
```

Comment:
you moved ApplyTableScanRedirection to last. why?

Reply:
Rule execution order doesn't matter, but I moved it back